pkg/lib/v0_2_0/keptn.go

```diff
@@ -83,3 +87,106 @@ func (k *Keptn) GetShipyard() (*Shipyard, error) {
 }
+
+// SendCloudEvent sends a cloudevent to the event broker
+func (k *Keptn) SendCloudEvent(event cloudevents.Event) error {
+	if k.UseLocalFileSystem {
+		log.Println(fmt.Printf("%v", string(event.Data())))
+		return nil
+	}
+
+	return k.EventSender.SendEvent(event)
+}
+
+// SendTaskStartedEvent sends a .started event for the incoming .triggered event the KeptnHandler was initialized with.
+// It returns the ID of the sent CloudEvent or an error
+func (k *Keptn) SendTaskStartedEvent(data keptn.EventProperties, source string) (string, error) {
+
+	outEventType, err := GetEventTypeForTriggeredEvent(k.CloudEvent.Type(), keptnStartedEventSuffix)
+	if err != nil {
+		return "", fmt.Errorf("could not determine .started event type for base event: %s", err.Error())
+	}
+
+	return k.sendEventWithBaseEventContext(data, source, err, outEventType)
+}
+
+// SendTaskStartedEvent sends a .status.changed event for the incoming .triggered event the KeptnHandler was initialized with.
+// It returns the ID of the sent CloudEvent or an error
+func (k *Keptn) SendTaskStatusChangedEvent(data keptn.EventProperties, source string) (string, error) {
+	outEventType, err := GetEventTypeForTriggeredEvent(k.CloudEvent.Type(), keptnStatusChangedEventSuffix)
+	if err != nil {
+		return "", fmt.Errorf("could not determine .started event type for base event: %s", err.Error())
+	}
+
+	return k.sendEventWithBaseEventContext(data, source, err, outEventType)
+}
+
+// SendTaskStartedEvent sends a .finished event for the incoming .triggered event the KeptnHandler was initialized with.
+// It returns the ID of the sent CloudEvent or an error
+func (k *Keptn) SendTaskFinishedEvent(data keptn.EventProperties, source string) (string, error) {
+	outEventType, err := GetEventTypeForTriggeredEvent(k.CloudEvent.Type(), keptnFinishedEventSuffix)
+	if err != nil {
+		return "", fmt.Errorf("could not determine .started event type for base event: %s", err.Error())
+	}
+
+	return k.sendEventWithBaseEventContext(data, source, err, outEventType)
+}
+
+func (k *Keptn) sendEventWithBaseEventContext(data keptn.EventProperties, source string, err error, outEventType string) (string, error) {
+	if source == "" {
+		return "", errors.New("must provide non-empty source")
+	}
+	keptnContext, err := k.CloudEvent.Context.GetExtension(keptnContextCEExtension)
+	if err != nil {
+		return "", fmt.Errorf("could not determine shkeptncontext of base event: %s", err.Error())
+	}
+
+	ce, err := k.createCloudEventWithContextAndPayload(outEventType, keptnContext, source, data)
+	if err != nil {
+		return "", fmt.Errorf("could not initialize CloudEvent: %s", err.Error())
+	}
+
+	if err := k.EventSender.SendEvent(*ce); err != nil {
+		return "", fmt.Errorf("could not send CloudEvent: %s", err.Error())
+	}
+	return ce.ID(), nil
+}
+
+func (k *Keptn) createCloudEventWithContextAndPayload(outEventType string, keptnContext interface{}, source string, data keptn.EventProperties) (*cloudevents.Event, error) {
```

Comment:
Please describe in a comment what this helper function is doing

Please also add inline comments, e.g., 
```
// use existing  sh keptn context for the new cloudevent
ce.SetExtension(keptnContextCEExtension, keptnContext)
```